### PR TITLE
Silent mode

### PR DESCRIPTION
**Pull request by Urvika-gola**
Adding silent mode feature.

### CODE REVIEW

File: res/layout/silent_mode.xml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            android:hint="HH"
+            android:text=""/>
+        <EditText
+            android:id="@+id/min1"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:inputType="number"
+            android:hint="MM"
+            android:text=""/>
+
```
Comment (JLouback) [inline res/layout/silent_mode.xml]: May seem trivial, but it's a good practice to have consistency in your code. In this case, decide whether to always have a new line between elements or always not to (I'd go for not like the first three inner elements here, but this is a matter of coding style and there's no wrong opinion)

File: res/values/strings.xml [hunk 2]
```diff
@@ -510,4 +510,21 @@
 <string name="sms_send">Send</string>
 <string name="sms_send_failure">SIP SMS sending failed</string>
 
+    <string name="silent_config">Configure Silent Mode</string>
+    <string name="silent_disable_enable">Disable/Enable</string>
+    <string name="silent_on">Silent Mode On</string>
+    <string name="silent_summary_off">Silent Mode is Off"</string>
```
Comment (JLouback) [inline res/values/strings.xml]: Consistency here too; either label "silent_summary_on" and "silent_summary_off" or both "silent_on" and "silent_off" (I always prefer shorter)

File: res/values/strings.xml [hunk 3]
```diff
@@ -510,4 +510,21 @@
...
 <string name="sms_send_failure">SIP SMS sending failed</string>
 
+    <string name="silent_config">Configure Silent Mode</string>
+    <string name="silent_disable_enable">Disable/Enable</string>
+    <string name="silent_on">Silent Mode On</string>
+    <string name="silent_summary_off">Silent Mode is Off"</string>
+    <string name="silent_add_days_hours">Add Work Days/Hours</string>
+    <string name="silent_optional">(Optional)</string>
+    <string name="silent_heading">To enable silent mode based on your schedule</string>
+    <string name="silent_24">1. Enter Time in 24 Hour Format </string>
```
Comment (JLouback) [inline res/values/strings.xml]: I think labels like "silent_enter_time" and "silent_enter_days" are more intuitive, what do you think?

File: res/layout/silent_mode.xml [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        <TextView
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="To"/>
```
Comment (JLouback) [inline res/layout/silent_mode.xml]: Is there a specific reason why all the other labels/text have a corresponding mapping in strings.xml and this one doesn't? Same for 'hour2' below

File: res/layout/silent_mode.xml [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+        <EditText
+            android:id="@+id/hour2"
```
Comment (JLouback) [inline res/layout/silent_mode.xml]: We might want to come up with better ids than "hour1" and "hour2". How about "hour_start" and "hour_end"? Same for min? You might have a better idea, this is just what comes to mind.

File: src/org/lumicall/android/preferences/SilentMode.java [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Locale;
+
+/**
+ * Created by urvikagola on 29/1/17.
+ */
+
+public class SilentMode extends Activity {
+    EditText hh1,mm1,hh2,mm2;
```
Comment (JLouback) [inline src/org/lumicall/android/preferences/SilentMode.java]: Similar comment to what I said above, hhStart, mmStart, hhEnd, mmEnd would be more legible.

**Comment by dpocock** [comment]
@Urvika-gola @JLouback thanks for getting this ready to merge

**Pull request merged**